tests/integration_tests/tests/agentless_tests/test_executions.py

```diff
@@ -818,25 +819,161 @@ def test_dry_run_execution(self):
-                                       skip_plugins_validation=True)
-        do_retries(verify_deployment_environment_creation_complete, 30,
-                   deployment_id=deployment_id)
+        scheduled_time = generate_scheduled_for_date()
+        execution = self.client.executions.start(deployment_id=dep_id,
+                                                 workflow_id='install',
+                                                 schedule=scheduled_time)
+        self._assert_execution_status(execution.id, 'scheduled')
+
+        time.sleep(62)  # Wait for exec to 'wake up'
+        self._assert_execution_status(execution.id, 'queued')
+        self.client.executions.update(snapshot_1.id, 'terminated')
+        self.wait_for_execution_to_end(execution)
+
+    def test_schedule_execution_and_create_snapshot_same_tenant(self):
+        """
+        Schedule an execution, then create snapshot.
+        Execution 'wakes up' while snapshot is still running, so it becomes
+        'queued' and start when snapshot terminates.
+        """
+        # The token in the container is invalid, create new valid one
+        create_api_token()
+        dsl_path = resource('dsl/sleep_workflows.yaml')
+        dep = self.deploy(dsl_path, wait=False, client=self.client)
+        dep_id = dep.id
+        do_retries(verify_deployment_env_created, 30, deployment_id=dep_id)
 
-        execution = self.client.executions.start(deployment_id=deployment_id,
+        scheduled_time = generate_scheduled_for_date()
+        execution = self.client.executions.start(deployment_id=dep_id,
                                                  workflow_id='install',
                                                  schedule=scheduled_time)
-        self.assertEquals('scheduled', execution.status)
-        # The execution was scheduled for 2 minutes from now
-        time.sleep(110)
-        # Make sure execution was executed
+        self._assert_execution_status(execution.id, 'scheduled')
+
+        # Create snapshot and keep it's status 'started'
+        snapshot = self._create_snapshot_and_modify_execution_status(
+            'started')
+
+        time.sleep(62)  # Wait for exec to 'wake up'
+        self._assert_execution_status(execution.id, 'queued')
+        self.client.executions.update(snapshot.id, 'terminated')
         self.wait_for_execution_to_end(execution)
+
+    def test_schedule_execution_while_execution_running_under_same_dep(self):
+        """
+        Start an execution and while it is running schedule an execution
+        for the future, under the same deployment.
+
+        """
+        # The token in the container is invalid, create new valid one
+        create_api_token()
+        dsl_path = resource('dsl/sleep_workflows.yaml')
+        dep = self.deploy(dsl_path, wait=False, client=self.client)
+        dep_id = dep.id
+        do_retries(verify_deployment_env_created, 30, deployment_id=dep_id)
```

Comment:
I think that the beginning of each test is similar, it can be in a common function